bezier_application/src/bezier_application.cpp

```diff
@@ -81,6 +83,18 @@ int main(int argc, char **argv)
   std::vector<bool> points_color_viz;
   std::vector<Eigen::Affine3d, Eigen::aligned_allocator<Eigen::Affine3d> > way_points_vector;
   std::vector<int> index_vector;
+  if(grinding_mode.compare("surface") == 0)
+  {
+    //Grinding on surface mode has been selected
+    bezier_planner.setSurfacingOn();
+  }else if (grinding_mode.compare("full") == 0){
+    //Apply the full grinding mode
+    bezier_planner.setSurfacingOff();
+  }else{
+    ROS_ERROR_STREAM("Unknown grinding mode selected." << std::endl <<
+        "Usage :" << std::endl << "roslaunch my_path_generator path_generator.launch meshname:=meshname.ply grinding:=full (or surface)");
+    return -1;
+  }
```

Comment:
Code formatting is not correct here.
Change the else if to an else so that anything but "on" will trigger the surfacing mode.